Sources/Implementation/DefaultDecisionService.swift

```diff
@@ -80,24 +67,26 @@ class DefaultDecisionService: OPTDecisionService {
     ///   - config: The project configuration containing experiment and feature details.
     ///   - experiment: The CMAB experiment to evaluate.
     ///   - user: The user context containing user ID and attributes.
+    ///   - bucketingId: User bucketing id
+    ///   - isAsync: Controls synchronous or asynchronous decision.
     ///   - options: Optional decision options (e.g., ignore user profile service).
     /// - Returns: A `CMABDecisionResult` containing the CMAB decisions( variation id, cmabUUID) with reasons
     
     private func getDecisionForCmabExperiment(config: ProjectConfig,
                                               experiment: Experiment,
                                               user: OptimizelyUserContext,
                                               bucketingId: String,
-                                              opType: OPType,
+                                              isAsync: Bool,
                                               options: [OptimizelyDecideOption]?) -> DecisionResponse<VariationDecision> {
         let reasons = DecisionReasons(options: options)
         guard let cmab = experiment.cmab else {
             logger.e("The experiment isn't a CMAB experiment")
             return DecisionResponse(result: nil, reasons: reasons)
         }
         
-        guard opType == .async else {
+        guard isAsync else {
```

Comment:
Can we add comments why we do not take the alternative solution:
1. check the current thread.
2. reject if running in main thread.
This solution will work fine but we're concerned this may be confusing when users call flag decision with cmab
- when called from main thread, we'll skip cmab with error logging
- when called from background thread, we'll include cmab result
So the decision can be different.